### PR TITLE
BUG: GH17525 Function _get_standard_colors resets global random seed

### LINKED ISSUE #17525: pandas.plotting.parallel_coordinates resets random seed!
#### Code Sample, a copy-pastable example if possible

```
>>> x = pd.DataFrame(np.zeros((10,10)))
>>> x
     0    1    2    3    4    5    6    7    8    9
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
2  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
3  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
4  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
5  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
6  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
7  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
8  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
9  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
>>> pandas.plotting.parallel_coordinates(x,0)
<matplotlib.axes._subplots.AxesSubplot object at 0x10b6b11d0>
>>> random.random()
0.25891675029296335
>>> pandas.plotting.parallel_coordinates(x,0)
<matplotlib.axes._subplots.AxesSubplot object at 0x10b6b11d0>
>>> random.random()
0.25891675029296335


```
#### Problem description

A call to the pandas.plotting.parallel_coordinates method resets the random seed

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.13.final.0
python-bits: 64
OS: Darwin
OS-release: 16.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: None.None

pandas: 0.20.3
pytest: 3.2.1
pip: 9.0.1
setuptools: 36.2.7
Cython: 0.26
numpy: 1.13.1
scipy: 0.19.1
xarray: None
IPython: 5.3.0
sphinx: 1.6.3
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.0.2
openpyxl: None
xlrd: 1.0.0
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.6.0
html5lib: 0.999999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by cmazzullo**
- [X] closes #17525 
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

File: pandas/plotting/_style.py [hunk 1]
```diff
@@ -114,8 +114,11 @@ def _get_standard_colors(num_colors=None, colormap=None, color_type='default',
             import random
 
             def random_color(column):
+                rstate = random.getstate()  # to avoid resetting the seed
```
Comment (gfyoung) [inline pandas/plotting/_style.py]: Point to the issue number.

Comment (gfyoung) [reply pandas/plotting/_style.py]: Also, since you're changing the behavior, let's document it.  A brief doc-string would be good here.

**Comment by gfyoung** [comment]
@cmazzullo : Also, you're definitely going to need tests.

**Comment by cmazzullo** [comment]
Okay, just added one.

File: pandas/plotting/_style.py [hunk 2]
```diff
@@ -114,8 +114,13 @@ def _get_standard_colors(num_colors=None, colormap=None, color_type='default',
             import random
 
             def random_color(column):
```
Comment (jreback) [inline pandas/plotting/_style.py]: I don't think ``random`` provides a context manger, maybe do this like:

```
rstate = random.getstate()
try:
    random.seed(...)
    return = [random...... ]
finally:
    random.setstate(rstate)
```

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/plotting/_style.py (hunk 2 above)
Comment (jreback) [reply pandas/plotting/_style.py]: actually just use this
```
In [20]: rs = pandas.core.common._random_state()

In [21]: rs.rand(3)
Out[21]: array([ 0.6786481 ,  0.94231001,  0.03334007])
```

File: doc/source/whatsnew/v0.21.0.txt [hunk 3]
```diff
@@ -614,6 +614,7 @@ Plotting
 - Line plots no longer assume monotonic x data when calculating xlims, they show the entire lines now even for unsorted x data. (:issue:`11310`, :issue:`11471`)
 - With matplotlib 2.0.0 and above, calculation of x limits for line plots is left to matplotlib, so that its new default settings are applied. (:issue:`15495`)
 - Bug in ``Series.plot.bar`` or ``DataFramee.plot.bar`` with ``y`` not respecting user-passed ``color`` (:issue:`16822`)
+- Bug in ``plotting._style._get_standard_colors`` resetting the random seed when generating random colors (:issue:`17525`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: reference the user visibile issue (e.g. parallel coordinates)?

File: pandas/tests/plotting/test_misc.py [hunk 4]
```diff
@@ -284,3 +284,20 @@ def test_subplot_titles(self):
                                                   title=title[:-1])
         title_list = [ax.get_title() for sublist in plot for ax in sublist]
         assert title_list == title[:3] + ['']
+
+    def test_get_standard_colors_random_seed(self):
+        """ For #17525 """
```
Comment (gfyoung) [inline pandas/tests/plotting/test_misc.py]: Generally, we just add a comment above, not a docstring.

File: doc/source/whatsnew/v0.21.0.txt [hunk 5]
```diff
@@ -614,6 +614,7 @@ Plotting
 - Line plots no longer assume monotonic x data when calculating xlims, they show the entire lines now even for unsorted x data. (:issue:`11310`, :issue:`11471`)
 - With matplotlib 2.0.0 and above, calculation of x limits for line plots is left to matplotlib, so that its new default settings are applied. (:issue:`15495`)
 - Bug in ``Series.plot.bar`` or ``DataFramee.plot.bar`` with ``y`` not respecting user-passed ``color`` (:issue:`16822`)
+- Bug in ``plotting._style._get_standard_colors`` causing ``plotting.parallel_coordinates`` to reset the random seed when using random colors (:issue:`17525`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: you don't need the first part, a user won't know what you are talking about.

**Review by jreback — APPROVED** [review]
lgtm with a minor comment. have't looked why its failing. ping on green.

File: pandas/plotting/_style.py [hunk 6]
```diff
@@ -111,11 +111,13 @@ def _get_standard_colors(num_colors=None, colormap=None, color_type='default',
...
-            import random
+            from pandas.core.common import _random_state
 
             def random_color(column):
-                random.seed(column)
-                return [random.random() for _ in range(3)]
+                """ Returns a random color represented as a list of length 3"""
+                # GH17525 use common._random_state to avoid resetting the seed
+                rs = _random_state()
+                return rs.rand(3)
```
Comment (jreback) [inline pandas/plotting/_style.py]: ``return rs.rand(3).tolist()`` will make this work.

**Review by jreback — APPROVED** [review]
lgtm. ping on green.

**Comment by jorisvandenbossche** [comment]
@cmazzullo Thanks!

**Comment by cmazzullo** [comment]
Sure thing

**Pull request merged**